sdk/azidentity/cache/cache.go

```diff
@@ -0,0 +1,47 @@
+//go:build go1.18 && (darwin || windows)
+// +build go1.18
+// +build darwin windows
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cache
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity/internal"
+	extcache "github.com/AzureAD/microsoft-authentication-extensions-for-go/cache"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/cache"
+)
+
+const defaultName = "msal.cache"
+
+func init() {
+	internal.NewCache = func(o *internal.TokenCachePersistenceOptions) (cache.ExportReplace, error) {
+		if o == nil {
+			return nil, nil
+		}
+		cp := *o
+		if cp.Name == "" {
+			cp.Name = defaultName
+		}
+		a, err := storage(cp)
+		if err != nil {
+			return nil, err
+		}
+		p, err := internal.CacheFilePath(cp.Name)
+		if err != nil {
+			return nil, err
+		}
+		return extcache.New(a, p)
+	}
+	internal.CacheFilePath = func(name string) (string, error) {
+		dir, err := cacheDir()
+		if err != nil {
+			return "", fmt.Errorf("couldn't create a cache file due to error %q", err)
```

Comment:
I suppose that on Linux, the lack of support will trip this error.  Still, it might be more clear to have an explicit error on Linux platforms stating that it's not supported.

Reply:
Hmm, but on Linux shouldn't the `storage()` and `cacheDir()` funcs not even exist?  I'm probably overlooking something, but the fact that the build succeeded on Linux in CI has me confused.

Reply:
I excluded this file on Linux with a build constraint.

Reply:
Oh yes I totally missed that.